src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(harbivore) {
+    if (harbivore instanceof Herbivore && !harbivore.hidden) {
+      harbivore.health -= 50;
+    }
+
+    if (harbivore.health <= 0) {
+      Animal.alive.splice(Animal.alive.indexOf(harbivore), 1);
```

Comment:
If the prey is already dead (not in `alive` array) then indexOf will return `-1` and the last element will be deleted. So, filter would be better here, try to do it with a filter

Reply:
Yeah, great remark. Thanks :boom:
I didn't use filter, but I added one more condition.